src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs

```diff
@@ -174,32 +174,45 @@ private void SkipType(DatabaseItemType type, SmugglerResult result)
 
         private SmugglerProgressBase.Counts ProcessIdentities(SmugglerResult result)
         {
-            using (var actions = _destination.Identities())
+            using (var clusterIdentityActions = _destination.ClusterIdentities())
             {
-                foreach (var kvp in _source.GetIdentities())
-                {
-                    _token.ThrowIfCancellationRequested();
-                    result.Identities.ReadCount++;
+                var sourceIdentities = _source.GetClusterIdentities();
+                WriteIdentities(result, sourceIdentities, clusterIdentityActions,
+                    () => result.ClusterIdentities.ReadCount++,
```

Comment:
If we have one type of identities, then maybe we can get rid of the lambdas here (maybe revert back to the old code?)